src/FEValues/FaceValues.jl

```diff
@@ -158,15 +158,15 @@ function Base.show(io::IO, d::MIME"text/plain", fv::FaceValues)
 end
 
 """
-    BCValues(func_interpol::Interpolation, geom_interpol::Interpolation, boundary_type::Union{Type{<:BoundaryIndex}})
+    BCValues([::Type{T},] func_interpol::Interpolation, geom_interpol::Interpolation, boundary_type::Union{Type{<:BoundaryIndex}})
 
-`BCValues` stores the shape values at all faces/edges/vertices (depending on `boundary_type`) for the geomatric interpolation (`geom_interpol`),
-for each dof-position determined by the `func_interpol`. Used mainly by the `ConstrainHandler`.
+`BCValues` stores the shape values at all faces/edges/vertices (depending on `boundary_type`) for the geometric interpolation (`geom_interpol`),
+for each dof-position determined by the `func_interpol`. Used mainly by the `ConstraintHandler`.
 """
-struct BCValues{T}
-    M::Array{T,3}
-    nqp::Array{Int}
-    current_entity::ScalarWrapper{Int}
+struct BCValues{V_GM, FQR} <: AbstractFaceValues
+    geo_mapping::V_GM # AbstractVector{GeometryMapping}
+    fqr::FQR          # FaceQuadratureRule
+    current_face::ScalarWrapper{Int}
```

Comment:
This should also give the possibility for boundary conditions involving normals, right?

Reply:
It would be possible to send the normal into the user-supplied function, so as long as all components are prescribed that would work. But, unfortunately, the normal is still not well-defined at cell vertices, so this would only be correct for smooth boundaries.

One way to implement this would be to store the prescribed dof, coordinate, and normal in a pre-calculated list, this would make it possible to define an averaged normal direction. But I'm not sure if this is wanted? (And would for sure be a different PR)